lib/ts/recipe/dashboard/api/user/create/passwordlessUser.ts

```diff
@@ -0,0 +1,117 @@
+import { APIInterface, APIOptions } from "../../../types";
+import STError from "../../../../../error";
+import Passwordless from "../../../../passwordless";
+import ThirdPartyPasswordless from "../../../../thirdpartypasswordless";
+import PasswordlessRecipe from "../../../../passwordless/recipe";
+import ThirdPartyPasswordlessRecipe from "../../../../thirdpartypasswordless/recipe";
+import { User } from "../../../../../types";
+import RecipeUserId from "../../../../../recipeUserId";
+import { parsePhoneNumber } from "libphonenumber-js/max";
+
+type Response =
+    | {
+          status: string;
+          createdNewRecipeUser: boolean;
+          user: User;
+          recipeUserId: RecipeUserId;
+      }
+    | {
+          status: "FEATURE_NOT_ENABLED_ERROR";
+      }
+    | {
+          status: "EMAIL_VALIDATION_ERROR";
+          message: string;
+      }
+    | {
+          status: "PHONE_VALIDATION_ERROR";
+          message: string;
+      };
+
+export const createPasswordlessUser = async (
+    _: APIInterface,
+    tenantId: string,
+    options: APIOptions,
+    __: any
+): Promise<Response> => {
+    let passwordlessRecipe: PasswordlessRecipe | undefined = undefined;
+
+    try {
+        passwordlessRecipe = PasswordlessRecipe.getInstanceOrThrowError();
+    } catch (_) {
+        try {
+            passwordlessRecipe = ThirdPartyPasswordlessRecipe.getInstanceOrThrowError().passwordlessRecipe;
```

Comment:
Using the `passwordlessRecipe` property within the `ThirdpartyPasswordlessRecipe` is important. If the user hasn't defined the validator function in the configuration in `ThirdpartyPasswordless`, these validator functions become undefined. However, even if the user has defined validators, they will be passed down to the passwordlessRecipe property. This behavior is detailed in the [code here](https://github.com/supertokens/supertokens-node/blob/master/lib/ts/recipe/thirdpartyemailpassword/recipe.ts#L128)